pandas-stubs/core/series.pyi

```diff
@@ -4632,7 +4627,26 @@ class Series(IndexOpsMixin[S1], NDFrame):
     ) -> Scalar: ...
     @overload
     def std(
-        self: Series[Timestamp] | Series[Timedelta],
+        self: Series[Never],
+        axis: AxisIndex | None = 0,
+        skipna: _bool | None = True,
+        ddof: int = 1,
+        numeric_only: _bool = False,
+        **kwargs: Any,
+    ) -> Any: ...
```

Comment:
similar comment - most people will use `std` on floating/int columns, so we can be safe here and make the result `float`